modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataInputDelegate.java

```diff
@@ -122,4 +127,18 @@ NamedTypeResolver getParameterResolver(String parameterName) {
       return failure(parameter.getType(), newFailure(e).onParameter(parameter.getName()));
     }
   }
+
+  private MetadataType adaptToListIfNecessary(MetadataType resolvedType, ParameterModel parameter,
```

Comment:
can't use `org.mule.runtime.api.meta.Typed` and unify this with the output resolving logic?

Reply:
Yes, but the reusable insignificant because `Output` case has more things to consider like: `Pagination`, `List<Result>`, Attributes Handling, etc